### PR TITLE
clustering: Add notes for GHE_EXTRA_SSH_OPTS

### PR DESCRIPTION
**Pull request by gnawhleinad**
Fixes https://github.com/github/backup-utils/issues/165.

/cc: @rubiojr, @electroniko

### CODE REVIEW

File: backup.config-example [hunk 1]
```diff
@@ -23,11 +23,13 @@ GHE_NUM_SNAPSHOTS=10
 #
 #GHE_RESTORE_HOST="github-standby.example.com"
 
-# Any extra options passed to the SSH command.  Nothing required by default
+# Any extra options passed to the SSH command.
+# In a single instance environment, nothing is required by default.
+# In a clustering environment, "-i path-to-ssh-private-key" is required.
```
Comment (electroniko) [inline backup.config-example]: I would just replace with "full-path-to-ssh-private-key" as somehow, relative paths are not working

**Comment by electroniko** [comment]
and yes, this will fix https://github.com/github/backup-utils/issues/165

**Comment by electroniko** [comment]
Also might be worth mentioning that any configurations on ssh (~/.ssh/config) regarding the cluster will have and adverse effect on running the tool

**Comment by gnawhleinad** [comment]
> Also might be worth mentioning that any configurations on ssh (~/.ssh/config) regarding the cluster will have and adverse effect on running the tool

Would that might be better in https://help.github.com/enterprise/admin/guides/installation/administrative-shell-ssh-access/?

**Comment by electroniko** [comment]
That I'm not sure, as the ssh config file will be something admins will do, to save time, let me recreate the cluster and try with both options enabled (the ssh config and the options in GHE_EXTRA_SSH_OPTS)

**Comment by gnawhleinad** [comment]
> That I'm not sure, as the ssh config file will be something admins will do

The existing [`README.md`](https://github.com/github/backup-utils/blob/master/README.md) has:

> Add the backup host's SSH key to the GitHub appliance as an *Authorized SSH key*. See dding an SSH key for shell access for instructions](https://enterprise.github.com/help/articles/adding-an-ssh-key-for-shell-access).

File: README.md [hunk 2]
```diff
@@ -83,6 +83,8 @@ download the most recent GitHub Enterprise version.
     host name. Additional options are available and documented in the
     configuration file but none are required for basic backup functionality.
 
+    * In a clustering environment, the `GHE_EXTRA_SSH_OPTS` key must be configured.
```
Comment (sbryant) [inline README.md]: `key must be configured.` -> ```must contain the `-i <private key file>` ssh option.```

might read a bit better and is more explicit about the requirements

Comment (gnawhleinad) [inline README.md]: :+1:!

**Comment by electroniko** [comment]
ok, tried with both config on ssh (~/.ssh/config) and on the options on the backup.config file, and it works:

```
 ./bin/ghe-backup 
Starting backup of backuptest-cluster.aws.octocluster.com in snapshot 20160218T180829
The authenticity of host 'backuptest-cluster.aws.octocluster.com (54.241.138.253)' can't be established.
ECDSA key fingerprint is SHA256:NTxZ0KYn1x4/9Ymp6IJUS9yjFANREUohyX8n6k0/zKs.
Are you sure you want to continue connecting (yes/no)? yes
Connect backuptest-cluster.aws.octocluster.com:122 OK (v2.5.0)
Backing up GitHub settings ...
Backing up SSH authorized keys ...
Backing up SSH host keys ...
Backing up MySQL database ...
Backing up Redis database ...
Warning: Permanently added '[backuptest-cluster-data-1]:122,[10.11.18.79]:122' (ECDSA) to the list of known hosts.
Backing up audit log ...
Backing up hookshot logs ...
Backing up Git repositories ...
* Starting backup for host: backuptest-cluster-data-0
* Starting backup for host: backuptest-cluster-data-1
* Starting backup for host: backuptest-cluster-data-2
Warning: Permanently added '[backuptest-cluster-data-0]:122' (ECDSA) to the list of known hosts.
Warning: Permanently added '[backuptest-cluster-data-1]:122' (ECDSA) to the list of known hosts.
Warning: Permanently added '[backuptest-cluster-data-2]:122' (ECDSA) to the list of known hosts.
Backing up GitHub Pages ...
* Starting backup for host: backuptest-cluster-data-0
* Starting backup for host: backuptest-cluster-data-1
* Starting backup for host: backuptest-cluster-data-2
Backing up asset attachments ...
* Starting backup for host: backuptest-cluster-data-0
* Starting backup for host: backuptest-cluster-data-1
* Starting backup for host: backuptest-cluster-data-2
Pruning 1 failed snapshot(s) ...

```

**Comment by electroniko** [comment]
so yeah, that one on the readme should be unrelated, so it is safe to merge as you had it on the doc, having some configuration on ssh does not affect the behavior.

**Pull request merged**